.circleci/config.yml

```diff
@@ -10,24 +30,82 @@ jobs:
     steps:
       - checkout
       - restore_caches
-      - setup_rust
-      - setup_system
+      - install_rust
+      - install_node_devlibs
       - print_current_versions
-
       - run:
-          name: Build using cargo make
+          name: Debug build and test using cargo make
           command: cargo make ci
+      - store_artifacts:
+          path: api_tests/log
+      - save_caches
 
+  release-build-test:
+    working_directory: ~/comit
+    machine:
+      image: ubuntu-1604:201903-01
+      docker_layer_caching: true
+    environment:
+      RUST_TEST_THREADS: "8"
+    steps:
+      - checkout
+      - restore_caches
+      - install_rust
+      - install_node_devlibs
+      - print_current_versions
+      - run:
+          name: Release build and test using cargo make
+          command: |
+            export CARGO_MAKE_CARGO_BUILD_TEST_FLAGS="--all --release"
+            export BTSIEVE_BIN=~/comit/target/release/btsieve
+            export CND_BIN=~/comit/target/release/cnd
+            cargo make ci
       - store_artifacts:
           path: api_tests/log
+      - run:
+          name: Consolidate release binaries
+          command: |
+            mkdir ~/artifacts
+            mv -v ~/comit/target/release/cnd ~/artifacts
+            mv -v ~/comit/target/release/btsieve ~/artifacts
+      - persist_to_workspace:
+          root: ~/artifacts
+          paths:
+            - cnd
+            - btsieve
 
-      - save_caches
+  publish-github-release:
+    working_directory: ~/comit
+    machine:
+      image: ubuntu-1604:201903-01
+      docker_layer_caching: true
+    environment:
+      RUST_TEST_THREADS: "8"
+    steps:
+      - attach_workspace:
+          at: ~/artifacts
+      - checkout
+      - restore_caches
+      - install_rust
+      - install_node_devlibs
+      - print_current_versions
+      - run:
+          name: "Package & Publish Release on GitHub"
+          command: |
+            go get github.com/tcnksm/ghr
+            VERSION=$(cargo pkgid -- cnd|cut -d# -f2)
+
+            # Let's abort if the git tag does not match the version in Cargo.toml as it would be fishy
```

Comment:
This is a very good idea!
Should we have that already as part of the release branch?
Like, make sure the version in the release branch matches the Cargo.toml version?
Even if not, I'd this the very first step, right after checkout: we don't need to do anything if those tags don't match :)

Another suggestions: The `pkgid` thingy looks a bit odd to me :sweat_smile: 
I had a go at the `cargo metadata` command and put together the following:

```
cargo metadata --no-deps --format-version 1 | jq '.packages[] | select(.name | contains("cnd")).version' -r
```

Gets the version of the `cnd` package and prints it on stdout.
Up to you if you want to use it :)

Reply:
... Pkgid is fine and documented in the cargo book along metadata

Reply:
Yes for the release branch

Reply:
> ... Pkgid is fine and documented in the cargo book along metadata

Hmm, I guess I just don't like this manual parsing of strings with `cut` and friends. As long as it is a documented and stable format, shouldn't matter much :)

Reply:
> > ... Pkgid is fine and documented in the cargo book along metadata
> 
> Hmm, I guess I just don't like this manual parsing of strings with `cut` and friends. As long as it is a documented and stable format, shouldn't matter much :)

Now I see your point.

Reply:
- I have added match against version in branch name
- I see your point for `pkgid`, you are welcome to do a follow PR on this.